src/types/capabilities.rs

```diff
@@ -31,21 +33,28 @@ use std::hash::Hash;
 pub struct Capabilities(
     // Note that this field isn't *actually* 'static.
     // Rather, it is tied to the lifetime of the `ZeroCopy` that contains this `Name`.
-    pub(crate) HashSet<&'static str>,
+    pub(crate) HashSet<Capability<'static>>,
 );
 
 impl Capabilities {
     /// Check if the server has the given capability.
-    pub fn has<S: ?Sized>(&self, s: &S) -> bool
-    where
-        for<'a> &'a str: Borrow<S>,
-        S: Hash + Eq,
-    {
+    pub fn has<'a>(&self, s: &Capability<'a>) -> bool {
```

Comment:
This is a backwards incompatible change, which is a bit unfortunate.. Maybe keep `has` with the old signature and have it call `has_str` + add a deprecation notice to use `has_str` instead? And call the method that takes a `Capability` `contains`?